src/relentless/simulate/hoomd.py

```diff
@@ -1228,11 +1240,27 @@ def act(self, timestep):
                         for i, j in self._rdf:
                             query_args = dict(self._rdf[i, j].default_query_args)
                             query_args.update(exclude_ii=(i == j))
+                            neighbors = (
+                                aabbs[j]
+                                .query(
+                                    snap.particles.position[type_masks[i]], query_args
+                                )
+                                .toNeighborList()
+                            )
+                            if snap.bonds is not None:
+                                bonds = numpy.vstack(
+                                    [self.bonds, numpy.flip(self.bonds, axis=1)]
+                                )
+                                # list intersect method from:
+                                # https://stackoverflow.com/a/67113105
+                                filter = ~(neighbors[:, None] == bonds).all(-1).any(1)
+                            else:
+                                filter = numpy.full((len(neighbors[:])), True)
                             # resetting when the samples are zero clears the RDF
                             self._rdf[i, j].compute(
                                 aabbs[j],
                                 snap.particles.position[type_masks[i]],
-                                neighbors=query_args,
+                                neighbors=neighbors.filter(filter),
```

Comment:
Then use just `neighbors` here:
```suggestion
                                neighbors=neighbors,
```